### PR TITLE
searcher: remove batch API v2

### LINKED ISSUE #23981: Remove batch API from searcher
Now that streaming is in, and seems stable, we should clean up the now-unused batch API.

### PR DESCRIPTION
**Pull request by camdencheek**
Now that the stream API is enabled by default, and appears to be stable, we can remove the unused batch API.

Each commit is pretty small and self-contained, so if you'd like to see the progression, review commit-by-commit. Note that all integration tests are now running against the streaming API.

Reopened from #24021
Closes #23981

### CODE REVIEW

File: internal/search/unindexed/unindexed.go [hunk 1]
```diff
@@ -133,14 +133,10 @@ func searchFilesInRepo(ctx context.Context, searcherURLs *endpoint.Map, repo typ
 	}
 
 	toMatches := newToMatches(repo, commit, &rev)
-
-	var onMatches func([]*protocol.FileMatch)
-	if stream != nil {
```
Comment (rvantonder) [inline internal/search/unindexed/unindexed.go:139]: How do we guarantee `stream` is non-nil? We still have some logic that sets stream to `nil` and then runs `r.resultsBatch`, which I think can end up running searcher (didn't check?): https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/graphqlbackend/search_results.go?L971-972&subtree=true

Do we guarantee that some other way here?

I tested with `(a and (b or c)) index:no` which should exercise that logic that sets stream to `nil` above but things still appear to work, so we guarantee this some other way?

File: internal/search/searcher/client.go [hunk 2]
```diff
@@ -197,54 +190,6 @@ func textSearchStream(ctx context.Context, url string, body []byte, cb func([]*p
 	return ed.LimitHit, err
 }
 
-func textSearch(ctx context.Context, url string, body []byte) ([]*protocol.FileMatch, bool, error) {
```
Comment (rvantonder) [inline internal/search/searcher/client.go:200]: 😮

**Review by rvantonder — APPROVED** [review]
This is great.

File: internal/search/searcher/client.go (hunk 2 above)
Comment (camdencheek) [reply internal/search/searcher/client.go:200]: this looks good, but it's really just removing code I duplicated a few lines above 😄

File: internal/search/unindexed/unindexed.go (hunk 1 above)
Comment (camdencheek) [reply internal/search/unindexed/unindexed.go:139]: Even when streaming is disabled, for each basic query, we pass in the [aggregator as the stream](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/search/run/aggregator.go?L127:9). Stream should never be `nil` in our backend functions, even if stream can be `nil` (right now) in the search resolver

**Pull request merged**